runtime/src/accounts_db.rs

```diff
@@ -5034,15 +5034,29 @@ impl AccountsDb {
                         let dirty_keys =
                             accounts_map.iter().map(|(pubkey, _info)| *pubkey).collect();
                         self.uncleaned_pubkeys.insert(*slot, dirty_keys);
+
+                        let infos: Vec<_> = accounts_map
+                            .iter()
+                            .map(|(pubkey, account_infos)| {
+                                account_infos
+                                    .iter()
+                                    .map(|(_, (store_id, stored_account))| {
+                                        (
+                                            pubkey,
+                                            AccountInfo {
+                                                store_id: *store_id,
+                                                offset: stored_account.offset,
+                                                stored_size: stored_account.stored_size,
+                                                lamports: stored_account.account_meta.lamports,
+                                            },
+                                        )
+                                    })
+                                    .collect::<Vec<_>>()
+                            })
+                            .collect();
```

Comment:
bit of a nit, but I thought of doing a `flatten` here. Might be slightly faster?

Reply:
It may be slightly faster with a flatten outside the lock here. But, it is about to be much faster with another modification in the next pr that builds on this. Each change should hopefully be easy to verify as correct.

Reply:
ah ok